server/pulp/plugins/util/publish_step.py

```diff
@@ -1046,36 +1069,36 @@ def cancel(self):
         self.downloader.cancel()
 
 
-class GetLocalUnitsStep(PluginStep):
+class SaveUnitsStep(PluginStep):
+    """
+    Any step that saves units to the database should use this step in order to ensure that
+    the repo unit counts are udpated properly.
+    """
+
+    def finalize(self):
+        repo_controller.rebuild_content_unit_counts(self.get_repo())
+
+
+class GetLocalUnitsStep(SaveUnitsStep):
     """
-    Given a list of unit keys, this will determine which ones are already in
-    pulp, and it will use the conduit to save each. This depends on there being
-    a parent step with attribute "available_units", which must be a list of
-    unit keys (which themselves are dictionaries).
+    Given an iterator of units, associate the ones that are already in Pulp with the
```

Comment:
Similar suggestion here to add a short, single sentence docstring above this great description here.  This is recommended style by PEP 257.